File: Week2 Assignment/sum_of_diagonals.py
#Sum of diagonals

# Input matrix
n = int(input("Enter size of square matrix (n x n): "))

print("Enter the matrix row-wise:")
matrix = [list(map(int, input().split())) for _ in range(n)]

# Initialize sums
main_diagonal = 0
secondary_diagonal = 0

for i in range(n):
    main_diagonal += matrix[i][i]
    secondary_diagonal += matrix[i][n - i - 1]

print("Sum of main diagonal:", main_diagonal)
print("Sum of secondary diagonal:", secondary_diagonal)
print("Total diagonal sum:", main_diagonal + secondary_diagonal)

File: Week2 Assignment/sum_of_diagonals.c
//Sum of diagonals

#include <stdio.h>

int main() {
    int n,i,j;
    printf("Enter the size of square matrix (n x n): ");
    scanf("%d", &n);

    int matrix[n][n];
    printf("Enter matrix elements:\n");
    for (i = 0; i < n; i++)
        for (j = 0; j < n; j++)
            scanf("%d", &matrix[i][j]);

    int mainDiagonal = 0, secondaryDiagonal = 0;

    for (i = 0; i < n; i++) {
        mainDiagonal += matrix[i][i];
        secondaryDiagonal += matrix[i][n - i - 1];
    }

    printf("Sum of main diagonal: %d\n", mainDiagonal);
    printf("Sum of secondary diagonal: %d\n", secondaryDiagonal);
    printf("Total diagonal sum: %d\n", mainDiagonal + secondaryDiagonal);

    return 0;
}

